### PR TITLE
[IMP] module payment_mercado_pago: I add two states in mapping constant.

### PR DESCRIPTION
**Pull request by maq-adhoc**
Although the lack of these statuses does not block the flow of a transaction, they create confusing records for the user.
The new states are:
 - Authorized: it is returned by Mercadopago when it authorizes payments by bundle. It should behave as a pending payment.
 - Cancelled: It is returned by Mercadopago when a payment is not made within the expected timeframe. This prevents odoo from leaving payment transactions in draft that cannot be cancelled by the customer.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

 - adhoc ticket = 68407
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/payment_mercado_pago/const.py
Comment (Demesmaeker) [inline addons/payment_mercado_pago/const.py:35]: Maybe map this status to 'error' then, and explicitly manage that case in `_process_notification_data` =]

**Review by Demesmaeker — CHANGES_REQUESTED** [review]
Hello,

Thanks for your work, small change needed, then will be good for me. :slightly_smiling_face: 

Have a nice day,

File: addons/payment_mercado_pago/const.py
Comment (maq-adhoc) [reply addons/payment_mercado_pago/const.py]: @Demesmaeker I am fine with adding the error status. 
Add human readable errors for the case of rejected. 
if you are ok with the change  I will do  rebase,

**Review by Demesmaeker — CHANGES_REQUESTED** [review]
Hello @maq-adhoc 
Sorry for the time it took me to come back to you, and thanks for your work here!
We discussed it a bit internally, and we agreed this is indeed what we want to have :partying_face: .
Small nitpicking though: could you make it so the lines don't go above 100 chars?
Thank you very much!
Have a nice day!

File: addons/payment_mercado_pago/const.py
Comment (AntoineVDV) [reply addons/payment_mercado_pago/const.py:33]: Indeed, `authorized` means that the funds are reserved and the payment method is valid; the ongoing operation should be allowed to continue (e.g. shipping).

Comment (AntoineVDV) [reply addons/payment_mercado_pago/const.py:33]: Hi @maq-adhoc 

> A payment with authorized status can change to status (done, cancel, rejected). It is necessary that the state of the transaction is pending.

An authorized payment can indeed change to `done`, but how could it be canceled or rejected without user input? To my understanding, _automatic_ capture payments are authorized right before they are captured.

Or maybe the user has a way to update the payment from MercadoPago's backend in between the authorization and the capture?

I'm not implying that what you suggest is impossible; I'm just trying to understand how MercadoPago differs from the other providers we have implemented so far :wink: 

I agree with the rest of your statements.

In the end, regardless of whether authorized payments can be canceled or not, I agree to flag them as `pending` because we should always receive a capture webhook notification anyway if/when they are captured. cc @Demesmaeker

**Comment by maq-adhoc** [comment]
> Hello @maq-adhoc
> 
> I'm closing this PR, as this'll be continued in #144804
> 
> Thanks for your work here (keeping your commit, just adapting the code ^^)

@Demesmaeker no problem. The beginning of the year didn't allow me to work on this. I had it scheduled for next week. If I can help in any way let me know.

Thanks

**Comment by Demesmaeker** [comment]
@maq-adhoc Should be good to go, just waiting for the final review ^^
A bit late, but have a nice year =]

**Pull request closed without merging**
